### PR TITLE
Adds "Upgrade" paywall to Workflow SMS features instead of disabling them

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

Adds "Upgrade" paywall to select options and use it for Workflow SMS actions instead of just disabling them when user does not have TEAMS plan.

Loom Video: https://www.loom.com/share/3cfbb090582d4dbba1f4eece7beb4350

Fixes: https://linear.app/calcom/issue/CAL-608/add-upgrade-paywall-to-workflow-sms-features-instead-of-disabling-them

**Environment**: Staging(main branch)

## How should this be tested?

- Test like shown in loom video

### CODE REVIEW

**Review by zomars — APPROVED** [review]
Thank you @CarinaWolli 🙏🏽

File: packages/ui/components/form/select/components.tsx [hunk 1]
```diff
@@ -52,7 +65,17 @@ export const OptionComponent = <
...
         props.isSelected && "dark:!bg-darkgray-300 !bg-neutral-900"
       )}>
-      <span>{props.label}</span> {props.isSelected && <Icon.FiCheck className="h-4 w-4" />}
+      <>
+        <span className="mr-auto">{props.label}</span>
+        {(props.data as unknown as ExtendedOption).needsUpgrade && (
+          <Tooltip content={t("upgrade_to_enable_feature")}>
+            <button type="button" onClick={() => router.replace("/teams")}>
+              <Badge variant="gray">{t("upgrade")}</Badge>
+            </button>
```
Comment (PeerRich) [inline packages/ui/components/form/select/components.tsx:74]: NIT: shouldnt this be a link instead? why a button with a router? 

`<Link href="/teams">` wouldve worked no?

Comment (PeerRich) [inline packages/ui/components/form/select/components.tsx:74]: also should we turn this into a component so we can reuse this in other places?

Comment (PeerRich) [reply packages/ui/components/form/select/components.tsx:74]: - Video Recordings @Udit-takkar 
- Disable Branding in settings

Comment (CarinaWolli) [reply packages/ui/components/form/select/components.tsx:74]: Should be `Link`, I'll change that

Comment (CarinaWolli) [reply packages/ui/components/form/select/components.tsx:74]: Yes would probably make sense, I'll create a new component for it. Will open up a PR for it tomorrow

**Pull request merged**